compiler/resolution/visibleFunctions.cpp

```diff
@@ -564,7 +534,7 @@ static void getVisibleMethodsImpl(const char* name, CallExpr* call,
       }
     }
 
-    if (block != rootModule->block) {
+    {
```

Comment:
> Moves the check for rootModule->block to be the first thing in
getVisibleMethodsImpl() and getVisibleFunctionsImpl(), to
safe unnecessary work.

This line and the addition on line 465 have to do with this part of your PR message, right?  I don't think this is doing what you think it's doing, if so.  The check here is to prevent us from getting the parent scope of a scope that doesn't have a parent scope.  rootModule has included things in the past - we've placed some code the compiler generates in there, if I remember right.  Have you gone through the contents of the rootModule carefully to be certain that there isn't anything in it any more?  Otherwise, I would be worried we're skipping symbols that should be considered.

At the very least, please don't leave random `{}` lying around, it clutters the code.

Reply:
Actually, I did. That's where the changes in createTaskFunction.cpp came from: because tiMarkHost is the only thing in rootModule right now.

I don't know how valuable it is to place functions in rootModule. Instead, we could place them in `theProgram`.

I will remove the {}.